### ISSUE TITLE
@pamaury : just a thought for better clarity / consistency with the [X.509 spec](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2), perhaps in a follow up PR we should name the `extensions` field here `private_extensions` since the spec states "_The X.509 v3 certificate format also allows communities to define private extensions to carry information unique to those communities._"  Thoughts?

### ISSUE THREAD

**Issue by timothytrippel** [lowRISC/opentitan#22620]
@pamaury : just a thought for better clarity / consistency with the [X.509 spec](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2), perhaps in a follow up PR we should name the `extensions` field here `private_extensions` since the spec states "_The X.509 v3 certificate format also allows communities to define private extensions to carry information unique to those communities._"  Thoughts?

_Originally posted by @timothytrippel in https://github.com/lowRISC/opentitan/pull/22573#discussion_r1569105674_

**Comment by pamaury** [comment]
Yes I think it makes sense. My only objection would be that `private_extension` is a bit long but if you think it's clearer then I am happy to change it.

**Comment by timothytrippel** [comment]
I think it helps differentiate from other types of extensions that would not be placed in the `extensions` list.
